api/docs/http_certificate_revocation_list.md

```diff
@@ -5,7 +5,11 @@ The `certificate_revocation_list` endpoint retrieves a Certificate Revocation Li
 from the master.  The master must be configured to be a CA.  The returned
 CRL is always in the `.pem` format.
 
-In all requests the `:environment` and `:nodename` must be given, but neither has any bearing on the request.
+Under Puppet Server's CA service, the `environment` parameter is ignored and can
+be omitted. Under a Rack or WEBrick Puppet master, `environment` is required and
+must be a valid environment, but it has no effect on the response.
+
+The `:nodename` is required, but has no effect on the response.
```

Comment:
Yeah, so with webrick puppet, using `ca` works:

```
[root@umofz4p9vzfypuk ~]# curl -k 'https://umofz4p9vzfypuk:8140/puppetn_list/ca?environment=production' --cert /etc/puppetlabs/puppet/ssl/certs/agent1.pem --key /etc/puppetlabs/puppet/ssl/private_keys/agent1.pem
-----BEGIN X509 CRL-----
MIICtzCBoAIBATANBgkqhkiG9w0BAQUFADA9MTswOQYDVQQDDDJQdXBwZXQgQ0E6
IHVtb2Z6NHA5dnpmeXB1ay5kZWxpdmVyeS5wdXBwZXRsYWJzLm5ldBcNMTUxMDIx
MjA1MDQ4WhcNMjAxMDE5MjA1MDQ5WqAvMC0wHwYDVR0jBBgwFoAUuyqAydNwFM2F
vQhEdy+gV+792SEwCgYDVR0UBAMCAQAwDQYJKoZIhvcNAQEFBQADggIBAFCcN2xf
p/mhI+X3+nmzsuuLwZ2rZEw0LvgYWyFzpWsrq5jK+Fb6QrPqA/hRJBsOK3ljrcvI
6hD8i61zlJDWX1BNrMcgjFmY5lQG1FaG7khvVLZVfF3MAGNbLTF/4ky+WeXpdapq
WJrWPKgs0L+WYOHFK6Lr3Aa4WyMVrM8WA8F4pik9enDKePrHzfagNvbe9mcvtTmP
WFWeom3pgv81ev46eKCKuO/lwPg0ZcR34/vuCGFyh3Y6X5ccg5q2u6zHme4OaoV8
/nWXKdMkxfOPcFIYrmCmawmT4/5C7w6SdBBwOXbO+zx6kErz39ht8wjM1SqGwcF4
3+b+gL99EjnxZDGEucckt56DELcQOUmr6SWwycIrHRc1xngoHR0JigPy/fitZCIo
xOgWjePfzPrnvmgjWoTuZkdM79S472LweN2goyFZKI8V/i9w05Z6dqwhmgCpItsh
th8CvNaD2Foa4/41GqvG6/luqciewrhI9obglcwFxnBVlMtAthYgTRnO4vcCLDWA
YcjLN/iGBmJJdu0BkyQqVj8A/pPitTVhAgkRpQ0PcD8CdTDPXVdQF8nVW+5Tf8VQ
sD6k1YLilykXwImHZbQw+a1kYhouIAuEql3GFioC7sUvKptKl4mzRodrQIcfQYR6
+O/11xblDjkDPrgVvlbPjyejZzlthHqVLcKU
-----END X509 CRL-----
```

But when using `foo` it doesn't:

```
[root@umofz4p9vzfypuk ~]# curl -k 'https://umofz4p9vzfypuk:8140/puppet-ca/v1/certificate_revocation_list/foo?environment=production' --cert /etc/puppetlabs/puppet/ssl/certs/agent1.pem --key /etc/puppetlabs/puppet/ssl/private_keys/agent1.pem
Forbidden request: agent1(10.32.126.102) access to /puppet-ca/v1/certificate_revocation_list/foo [find] authenticated  at :123
```

It may be that puppetserver behaves differently?